plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPageModel.java

```diff
@@ -472,7 +472,7 @@ public void setRoute(IRoute newRoute) {
 				this.deployProject, this.projects, 
 				this.sourcePath, this.podPath, this.useInferredPodPath,
 				getService(), getServiceItems(),
-				route, this.selectDefaultRoute, this.routesByProject,
+				newRoute, this.selectDefaultRoute, this.routesByProject,
```

Comment:
There is method updateRoutes() that is called before updateRoute(). The right place to narrow the list of routes is updateRoutes(). Now its implementation calls getAllRoutes(service) that returns all routes for the project. @adietish , could you please confirm if this logic is correct?